torchtitan/experiments/torchcomms/parallel_dims.py

```diff
@@ -49,81 +49,125 @@ def _calculate_ranks_per_dimension(
     return ranks_per_dim
 
 
-@dataclass
-class TorchCommsParallelDims(ParallelDims):
-    def _build_mesh_without_ep(self) -> DeviceMesh:
-        # TODO: support EP
-        dims = []
-        names = []
-        for d, name in zip(
-            [self.pp, self.dp_replicate, self.dp_shard, self.cp, self.tp],
-            ["pp", "dp_replicate", "dp_shard", "cp", "tp"],
-        ):
-            if d > 1:
-                dims.append(d)
-                names.append(name)
+def _create_device_mesh(
+    world_size: int,
+    mesh_shape: tuple,
+    mesh_dim_names: List[str],
+) -> tuple:
```

Comment:
A better way to return here is:
Option1: return a dict so key will be str and value will be whatever you are returning here.
Option2: return an object

Directly returning tuple is hacky..